pyop2/base.py

```diff
@@ -436,6 +447,14 @@ def _is_indirect_and_not_read(self):
         return self._is_indirect and self._access is not READ
 
     @property
+    def _is_readonly(self):
+        return self._access == READ
+
+    @property
+    def _is_written(self):
```

Comment:
well, we distinguish between "is_written" and "is_incremented", so isn't it better and much clearer to have properties for each case? as we also have now, incidentally